src/headset/oculus.c

```diff
@@ -0,0 +1,531 @@
+#include "headset/headset.h"
+#include "event/event.h"
+#include "graphics/graphics.h"
+#include "graphics/texture.h"
+#include "lib/vec/vec.h"
+#include "math/mat4.h"
+#include "math/quat.h"
+
+#include <stdbool.h>
+#include <OVR_CAPI.h>
+#include <OVR_CAPI_GL.h>
+
+typedef struct {
+  bool isInitialized;
+  bool isRendering;
+  bool isMirrored;
+  bool hmdPresent;
+  headsetRenderCallback renderCallback;
+  vec_controller_t controllers;
+  ovrSession session;
+  ovrGraphicsLuid luid;
+  ovrTrackingState ts;
+  ovrInputState is;
+  int frameIndex;
+  float clipNear;
+  float clipFar;
+  int lastButtonState;
+  struct {
+    ovrSizei size;
+    GLuint fboId;
+    GLuint depthId;
+    ovrTextureSwapChain chain;
+  } eyeTextures[2];
+  ovrMirrorTexture mirrorTexture;
+  GLuint mirrorFBO;
+} HeadsetState;
+
+static HeadsetState state;
+
+void lovrHeadsetInit() {
+  vec_init(&state.controllers);
```

Comment:
This is here and on 67, can probably just remove this

Reply:
woops